nalgebra-sparse/src/csr.rs

```diff
@@ -184,62 +185,33 @@ impl<T> CsrMatrix<T> {
         num_rows: usize,
         num_cols: usize,
         row_offsets: Vec<usize>,
-        col_indices: Vec<usize>,
-        values: Vec<T>,
+        mut col_indices: Vec<usize>,
+        mut values: Vec<T>,
     ) -> Result<Self, SparseFormatError>
     where
-        T: Scalar,
+        T: Scalar + Zero,
```

Comment:
I guess the `Zero` bound comes from needing to possible construct a buffer with e.g. `Vec::<T>resize`, is this correct?

Reply:
Yes, then the buffer needs to be filled with some values, in this case with zeros.